### PR TITLE
fix: minor insights v2 tweaks

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23248?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2124

## **Manual testing steps**

1. Go to https://metamask.github.io/test-dapp
2. Connect your Flask Wallet
3. Call "Malicious ERC20 transfer"
4. Confirm the transaction.

### **Before**
beforeDark
beforeLight

### **After**
afterLight
afterDark

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-delineator/snap-delineator.js
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-delineator/snap-delineator.js:86]: The issue asks for the color to be `textDefault`, perhaps always? @eriknson

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (FrederikBolding) [inline ui/components/app/snaps/insight-warnings/insight-warnings.js:165]:
```suggestion
              <Text as="span" key={action] fontWeight={FontWeight.Medium}>
                {stripHttpSchemes(origin)}
              </Text>
```

Same suggestion applies elsewhere

**Comment by FrederikBolding** [comment]
@Jonathansoufer there also is a linting problem with the translations file and you will need to sign the CLA as this is your first PR 😄

File: ui/components/app/snaps/snap-delineator/snap-delineator.js
Comment (Jonathansoufer) [reply ui/components/app/snaps/snap-delineator/snap-delineator.js:86]: Visually, this would be the effect. Waiting on @eriknson for confirmation.

Screenshot 2024-03-07 at 10 20 51
Screenshot 2024-03-07 at 10 19 48

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (FrederikBolding) [inline ui/components/app/snaps/insight-warnings/insight-warnings.js:164]:
```suggestion
              <Text as="span" key={action} fontWeight={FontWeight.Medium}>
```
I believe we usually use Medium for these

File: app/_locales/en/messages.json
Comment (FrederikBolding) [inline app/_locales/en/messages.json:2110]:
```suggestion
```

File: ui/components/app/snaps/snap-delineator/snap-delineator.js
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-delineator/snap-delineator.js:86]: I'm just now realizing this is a change for the delineator. This would change it in a bunch of places.

Is that the intention @eriknson ?

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (Jonathansoufer) [reply ui/components/app/snaps/insight-warnings/insight-warnings.js:164]: Fixed, although I think the difference is too little to bring awareness. But again, it's up to @eriknson 

Screenshot 2024-03-07 at 10 56 59
Screenshot 2024-03-07 at 10 56 23

**Comment by Jonathansoufer** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/_locales/en/messages.json
Comment (FrederikBolding) [inline app/_locales/en/messages.json:2127]:
```suggestion
```

**Review by FrederikBolding — COMMENTED** [review]
@Jonathansoufer You need to update some snapshots in the Jest testing it seems

File: ui/components/app/snaps/snap-delineator/snap-delineator.js
Comment (FrederikBolding) [reply ui/components/app/snaps/snap-delineator/snap-delineator.js]: I confirmed this with Erik

File: ui/components/app/snaps/snap-delineator/snap-delineator.js
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-delineator/snap-delineator.js:86]:
```suggestion
            color={TextColor.textDefault}
```

File: ui/pages/confirmations/components/security-provider-banner-alert/blockaid-banner-alert/__snapshots__/blockaid-banner-alert.test.js.snap
Comment (FrederikBolding) [inline ui/pages/confirmations/components/security-provider-banner-alert/blockaid-banner-alert/__snapshots__/blockaid-banner-alert.test.js.snap:54]: This seems like an unexpected change, I think it should be reverted

**Comment by eriknson** [comment]
This modal component is deprecated so let's use the new one like we do [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/snaps/snap-install-warning/snap-install-warning.js#L28)

**Review by Jonathansoufer — COMMENTED** [review]
All comments addressed

**Review by FrederikBolding — DISMISSED** [review]

File: ui/pages/confirmations/send/send-content/__snapshots__/send-content.component.test.js.snap
Comment (FrederikBolding) [inline ui/pages/confirmations/send/send-content/__snapshots__/send-content.component.test.js.snap:291]: Not sure what this is, can most likely be reverted?

Comment (Jonathansoufer) [reply ui/pages/confirmations/send/send-content/__snapshots__/send-content.component.test.js.snap:291]: this came due the merge from main and then updating the snapshots.

**Review by FrederikBolding — DISMISSED** [review]

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (GuillaumeRx) [inline ui/components/app/snaps/insight-warnings/insight-warnings.js:133]: `warning.length` should be passed into the translation variables so that translators have more context when translating

Comment (GuillaumeRx) [reply ui/components/app/snaps/insight-warnings/insight-warnings.js:133]: My bad, you're right !

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (Jonathansoufer) [reply ui/components/app/snaps/insight-warnings/insight-warnings.js]: Good hint. Changed.

File: ui/components/app/snaps/insight-warnings/insight-warnings.js
Comment (FrederikBolding) [inline ui/components/app/snaps/insight-warnings/insight-warnings.js:105]: Header should be outside the body

Comment (Jonathansoufer) [reply ui/components/app/snaps/insight-warnings/insight-warnings.js]: Good catch. Fixed

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**
